pkgs/tools/networking/pihole-ftl/default.nix

```diff
@@ -0,0 +1,50 @@
+{ cmake
+, fetchFromGitHub
+, gmp
+, lib
+, libidn
+, nettle
+, readline
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pihole-ftl";
+  version = "5.7";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "FTL";
+    rev = "v${version}";
+    sha256 = "1xls9s4ss1dxn1n5n17amanqqwnsqd5x7idkwr06z1jxnx2rf2b6";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ gmp libidn nettle readline ];
+
+  postPatch = ''
+    substituteInPlace src/CMakeLists.txt \
+      --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX"
+
+    substituteInPlace src/version.h.in \
+      --replace "@GIT_VERSION@" "v${version}" \
+      --replace "@GIT_DATE@" "2020-12-26" \
+      --replace "@GIT_BRANCH@" "master" \
+      --replace "@GIT_TAG@" "v${version}" \
+      --replace "@GIT_HASH@" "builtfromreleasetarball"
+  '';
+
+  installPhase = ''
```

Comment:
Please add `runHook preInstall` at the beginning, and `runHook postInstall` at the end. Otherwise, people who override this package and attempt to use `preInstall` or `postInstall` phases might be confused when nothing happens.